DbgProvider/Debugger.psm1

```diff
@@ -961,6 +961,13 @@ function ~x
                 {
                     Get-DbgUModeThreadInfo -Current
                 }
+                elseif($threadId -match '~\[\s*([0-9a-f]{1,4})\s*\]') {
+                    [Uint32] $TID = 0
+                    if( ![UInt32]::TryParse($matches[1], [System.Globalization.NumberStyles]::HexNumber, $null, [ref] $TID ) ) {
+                        throw "Invalid TID: $TID"
```

Comment:
>$TID [](start = 44, length = 4)

This should be something like "`throw "Invalid TID: $matches[1]"`" (because if the parse failed, `$TID` will be zero).

Also please match the existing brace style.

Reply:
I have addressed these points as well.
However, I believe it is impossible for the `throw` statement to ever be reached because the `elseif ... -match ...[0-9-a-f]` statement wouldn't pass an imparsable (non hex) `$matches[1]`.

Reply:
Good point. I still like this way better, because it seems to have a better chance of having useful information if we fumbled the regex or something.

---
In reply to: [601133813](https://github.com/microsoft/DbgShell/pull/96#discussion_r601133813) [](ancestors = 601133813)